### PR TITLE
Port source-build patches to dev/release/2.0

### PR DESCRIPTION
**Pull request by nguerrera**

### CODE REVIEW

**Comment by nguerrera** [comment]
dotnet/source-build#250 in progress against this. WIP until green on both sides.

File: build_projects/dotnet-cli-build/UpdateVersionsRepo.cs [hunk 1]
```diff
@@ -3,9 +3,12 @@
 
 using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities;
-using Microsoft.DotNet.VersionTools.Automation;
 using System.IO;
 
+#if !SOURCE_BUILD
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/UpdateVersionsRepo.cs]: Would it be better to just #if out this whole task, and then not execute the task as well?  That's the approach used elsewhere.

File: build_projects/update-dependencies/update-dependencies.sh [hunk 2]
```diff
@@ -30,26 +30,28 @@ if [ -z "${DOTNET_INSTALL_DIR:-}" ]; then
...
-if [ $? -ne 0 ]; then
-    echo "Failed to install stage 0"
-    exit 1
+if [ -z "$DOTNET_TOOL_DIR" ]; then
+    $REPO_ROOT/scripts/obtain/dotnet-install.sh -Channel master -Architecture x64
+
+    if [ $? -ne 0 ]; then
+        echo "Failed to install stage 0"
+        exit 1
+    fi
```
Comment (eerhardt) [inline build_projects/update-dependencies/update-dependencies.sh]: (nit) the `fi` spacing is off.  It shouldn't have spaces in front of it.

File: src/tool_fsharp/tool_fsc.csproj [hunk 3]
```diff
@@ -13,6 +13,8 @@
     <PackageReference Include="Microsoft.NetCore.App" Version="$(CLI_SharedFrameworkVersion)" />
     <PackageReference Include="Microsoft.FSharp.Compiler" Version="$(CLI_FSharp_Version)" />
     <PackageReference Include="CliDeps.Satellites.FSharp" Version="$(CLI_FSharp_Satellites_Version)" />
+    <PackageReference Include="System.Reflection.Metadata" Version="1.4.2" />
```
Comment (eerhardt) [inline src/tool_fsharp/tool_fsc.csproj]: I'm not sure I understand these two changes.  Do you know why they are necessary?

File: build/BundledRuntimes.props [hunk 4]
```diff
@@ -80,21 +80,21 @@
     </_DownloadAndExtractItem>
 
   <_DownloadAndExtractItem Include="AspNetCoreRuntimeInstallerWixLibFile"
-                           Condition=" '$(AspNetCoreRuntimeInstallerWixLibFile)' != '' And !Exists('$(AspNetCoreRuntimeInstallerWixLibFile)')">
+                           Condition="'$(IncludeAspNetCoreRuntime)' == 'true' And '$(SkipBuildingInstallers)' != 'true' And '$(AspNetCoreRuntimeInstallerWixLibFile)' != '' And !Exists('$(AspNetCoreRuntimeInstallerWixLibFile)')">
```
Comment (livarcocc) [inline build/BundledRuntimes.props]: We are going to have to port these up to core-sdk. We should probably file an issue there and link to this PR.

File: src/tool_fsharp/tool_fsc.csproj (hunk 3 above)
Comment (nguerrera) [reply src/tool_fsharp/tool_fsc.csproj]: There's a nuget downgrade warning as error in the f# graph without it. I couldn't figure out why it's only in source build, though. This is in the patches too.

File: build_projects/dotnet-cli-build/UpdateVersionsRepo.cs (hunk 1 above)
Comment (nguerrera) [reply build_projects/dotnet-cli-build/UpdateVersionsRepo.cs]: Fixing. The patches did it this way and then I adjusted the others without fixing this one.

File: build/BundledRuntimes.props (hunk 4 above)
Comment (nguerrera) [reply build/BundledRuntimes.props]: I'll take care of it as soon as this is merged

**Pull request merged**